parseAndPopulate/capability.py

```diff
@@ -126,51 +132,54 @@ def __init__(self, log_directory: str, hello_message_file: str, prepare: Prepare
         # Vendor modules send from API
         if self.api and not sdo:
             self.platform_data = []
-            json_file = open(hello_message_file.split('.xml')[0] + '.json')
-            impl = json.load(json_file)
+            with open('{}.json'.format(hello_message_file.split('.xml')[0]), 'r') as f:
+                impl = json.load(f)
             self.initialize(impl)
-            json_file.close()
         # Vendor modules loaded from directory
         if not self.api and not sdo:
-            if os.path.isfile('/'.join(self.split[:-1]) + '/platform-metadata.json'):
+            path = '/'.join(self.split[:-1])
+            if os.path.isfile('{}/platform-metadata.json'.format(path)):
                 self.platform_data = []
-                json_file = open('/'.join(self.split[:-1]) + '/platform-metadata.json')
-                platforms = json.load(json_file)['platforms']['platform']
-                for impl in platforms:
-                    self.initialize(impl)
-                json_file.close()
+                with open('{}/platform-metadata.json'.format(path), 'r') as f:
+                    platforms = json.load(f)['platforms']['platform']
+                for implementation in platforms:
+                    self.initialize(implementation)
             else:
                 self.platform_data = []
                 LOGGER.debug('Setting metadata concerning whole directory')
                 self.owner = 'YangModels'
                 self.repo = 'yang'
-                repo = repoutil.RepoUtil('{}{}/{}'.format(github_url, self.owner, self.repo), self.logger)
-                repo.clone()
+                repo_url = '{}{}/{}'.format(github_url, self.owner, self.repo)
+                repo = repoutil.load(self.yang_models_dir, repo_url)
                 self.path = None
                 self.branch = 'master'
                 self.branch = repo.get_commit_hash(self.branch)
                 repo.remove()
                 # Solve for os-type
-                if 'nx' in self.split[4]:
```

Comment:
why did we remove split[4]. please show me or remind me what split holds. if I remember correctly the string itself could contain more then just nx in split[4] that is why I used 'nx' in self.split[4].

your way we are just looking if the list contains string 'nx' exactly as is

Reply:
self.split holds splitted path to hello_message_file by '/' character
I think that expected path is something like yangmodels/yang/vendors/cisco/nx/ ...
However, if (for example) absolute path is passed, then 'nx' wil not be on split[4]
So if we are able to get position of 'nx' (= platform name), then we can get indexes of vendors, software-type ...